src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md

```diff
@@ -58,15 +57,17 @@ Memory consumption heavily depends on your test scenarios. To estimate the memor
 run the test on your development machine with 100VUs and multiply the consumed memory by the target number of VUs.
 
 Simple tests will use ~1-5MB per VU. (1000VUs = 1-5GB).
-Tests that are using file uploads can consume tens of megabytes per VU.
+Tests that are using file uploads, or load large JS modules, can consume tens of megabytes per VU.
+Keep in mind that each VU has a copy of all JS modules your test uses.
+If you need to share memory between VUs, consider using [SharedArray](/javascript-api/k6-data/sharedarray/), or an external data store, such as [Redis](/javascript-api/k6-experimental/redis/).
 
 ## General advice
 
 ### Make your test code resilient
 
 When running large stress tests, your script can't assume anything about the HTTP response.
 Often performance tests are written with a "happy path" in mind.
-For example, a "happy path" check like the one below is something that we see in k6 often.
+For example, a "happy path" check like the one below is something that we often see in k6 scripts.
```

Comment:
Now that I understand what "Happy path" is, is this really a happy path check? It seems like it does not assume the happy path, since it accepts the possibility of an incorrect body size. 

Instead, it seems like it's more a case of error handling that's not general enough.

Proposed change:

```
Even if scripts accept the SUT won't always respond according to the happy path, it still might not handle a sufficiently wide range of errors. 
For example, the following check is something we often see in k6 scripts:

Reply:
It is a happy path check, since it doesn't take into account failure conditions under load, when the response body wouldn't exist. The body size check is a standard assertion, and the user might want the test to fail if it's incorrect.

Your suggestion sounds more confusing, sorry, so I'd rather leave this as is.

Reply:
> It is a happy path check, since it doesn't take into account failure conditions under load, when the response body wouldn't exist. The body size check is a standard assertion, and the user might want the test to fail if it's incorrect.

I don't understand. Isn't an incorrect body size another indication of system failure? Or are we defining failure as "no body response at all?". From the Wikipedia page, I thought a happy path is where there are no error conditions.